index.js

```diff
@@ -124,8 +124,16 @@ module.exports = {
   included(app) {
     this._super.included.apply(this, arguments)
 
-    if (typeof app.import !== 'function' && app.app) {
-      app = app.app
+    // If we have the _findHost() method (in ember-cli >= 2.7.0), we'll just use that.
+    if (typeof this._findHost === 'function') {
```

Comment:
Seems like Ember should make a public API of this.

Reply:
@robmadole you are probably right and while the private API does work for this I went ahead and pulled it out. Just using the loop instead. Deeply nested addons are a bit of an issue, but this seems to be the accepted workaround for it.